### PR TITLE
Refactoring of new raid system

### PR DESCRIPTION
**Pull request by daanschipper**
Added loads of comments to understand better what is happening. 
Removed not useful code (at least to my understanding, no testing is available)
Rewritten some parts
Added TODOs

### CODE REVIEW

File: helpers/raids.js [hunk 1]
```diff
@@ -258,15 +246,10 @@ async function addRaid(msg, boss) {
...
-  // check if array contains the word mystic
-  let isMystic = textArray.indexOf("mystic") + textArray.indexOf(mysticemoji);
-  if (isMystic >= 0) { // if yes
-    // remove the element from the array
-    textArray.splice(isMystic, 0);
-  }
-  // if gym is blue this is > 0 otherwise 0
-  ++isMystic;
+  // check if gym is mystic
+  let isMystic = /<:mystic:\d*>/.test(msg)
```
Comment (gvlekke) [inline helpers/raids.js]: you are now only testing for emoji? and not for the text 'mystic', right?

**Comment by gvlekke** [comment]
for other changes I thing its oke, but did not test the functionality. Maybe someone else can do that?  Dont know if i have time this weekend

File: helpers/raids.js (hunk 1 above)
Comment (daanschipper) [reply helpers/raids.js]: That's true, I thought that by appending the mystic emoji to the raid you were saying the gym is mystic. The regex can easily be changed to /mystic/ to match any form of mystic in the message if that is preffered

Comment (gvlekke) [reply helpers/raids.js]: Mmm I'm actually unaware if it's necessary.

Comment (daanschipper) [reply helpers/raids.js]: This functionality will be replaced by the emoji reactions anyway, so I don't think this is that important

**Comment by daanschipper** [comment]
Alright, some of the old functionality is not present yet, but that is not present in the branch raid_final. Updating of the raid message does not update a message, but sends a new message. I'll look into that

**Comment by daanschipper** [comment]
refactoring needs more work than expected, i'll close this pr for now

**Pull request closed without merging**